packages/components/src/MaskedInput.tsx

```diff
@@ -234,7 +234,7 @@ const MaskedInput = React.forwardRef<HTMLInputElement, MaskedInputProps>(
       checkValue: string,
       cursorPosition = checkValue.length
     ): boolean {
-      const patternRegex = new RegExp(pattern);
+      const patternRegex = new RegExp(`^${pattern}$`);
```

Comment:
Good catch... side note I wonder if we should just allow the regex pattern to be passed in instead of just taking a string, or if we should just include the `^` and `$` in the string that's passed in instead of implicitly adding it here... but I think it's fine

Reply:
I wasn't sure about that too, ended up updating it here to fix the issues with all dependent components at once. If this doesn't work for some use case we can change it in the future.